roles/ec2-app/tasks/elasticache.yml

```diff
@@ -42,15 +42,19 @@
 
   when: ec_data.ec_sg is defined
 
+- name: Calculate identifier length
+  set_fact:
+    ecache_id_len: "{{  [ec_data.ec_name, opg_data.stack ] | join('-') | length  }}"
+
 - name: generate cluster identifier
   set_fact:
     ecc_instance_name: "{{ [ec_data.ec_name, opg_data.stack] | join('-') }}"
-  when: "{{ [ec_data.ec_name, opg_data.stack ] | join('-') | length <= 20}}"
+  when: ecache_id_len <= 20
 
 - name: generate cluster identifier
```

Comment:
Same naming for these 2 tasks as above to be used here? e.g. `generate truncated cluster identifier` `